pkg/generate/app/cmd/newapp.go

```diff
@@ -662,7 +662,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 			switch {
 			case refInput.ExpectToBuild:
 				glog.V(4).Infof("will add %q secrets into a build for a source build of %q", strings.Join(c.Secrets, ","), refInput.Uses)
-				if err := refInput.Uses.AddBuildSecrets(c.Secrets); err != nil {
+				if err := refInput.Uses.AddBuildSecrets(c.Strategy, c.Secrets); err != nil {
```

Comment:
is c.Strategy sufficient here?  c.Strategy comes form the command line flags, right?  But you can also end up in a docker strategy path depending on what's in your source repository.  How does this behave with, for example:
oc new-app https://github.com/openshift/ruby-hello-world (plus injection argument)?